src/main/java/com/azure/cosmos/examples/changefeed/SampleChangeFeedEstimator.java

```diff
@@ -0,0 +1,217 @@
+        try {
+            //Summary of the next four commands:
+            //-Create an asynchronous Azure Cosmos DB client and database so that we can issue async requests to the DB
+            //-Create a "feed container" and a "lease container" in the DB
+            logger.info("Create CosmosClient");
+            CosmosAsyncClient client = getCosmosClient();
+
+            logger.info("Create sample's database: " + DATABASE_NAME);
+            CosmosAsyncDatabase cosmosDatabase = createNewDatabase(client, DATABASE_NAME);
+
+            logger.info("Create container for documents: " + COLLECTION_NAME);
+            CosmosAsyncContainer feedContainer = createNewCollection(client, DATABASE_NAME, COLLECTION_NAME);
+
+            logger.info("Create container for lease: " + COLLECTION_NAME + "-leases");
+            CosmosAsyncContainer leaseContainer = createNewLeaseCollection(client, DATABASE_NAME, COLLECTION_NAME + "-leases");
+
+            //Model of a worker thread or application which leases access to monitor one or more feed container
+            //partitions via the Change Feed. In a real-world application you might deploy this code in an Azure function.
+            //The next line causes the worker to create and start an instance of the Change Feed Processor. See the implementation of getChangeFeedProcessor() for guidance
+            //on creating a handler for Change Feed events. In this stream, we also trigger the insertion of 10 documents on a separate
+            //thread.
+            logger.info("Start Change Feed Processor on worker (handles changes asynchronously)");
+            ChangeFeedProcessor changeFeedProcessorMainInstance = new ChangeFeedProcessorBuilder()
+                .hostName("SampleHost_1")
+                .feedContainer(feedContainer)
+                .leaseContainer(leaseContainer)
+                .handleChanges(handleChangesWithLag())
+                .buildChangeFeedProcessor();
+
+            try {
+                changeFeedProcessorMainInstance
+                    .start()
+                    .subscribeOn(Schedulers.boundedElastic())
+                    .timeout(Duration.ofSeconds(10))
+                    .then(Mono.just(changeFeedProcessorMainInstance)
+                              .delayElement(Duration.ofSeconds(10))
+                              .flatMap(value -> changeFeedProcessorMainInstance
+                                  .stop()
+                                  .subscribeOn(Schedulers.boundedElastic())
+                                  .timeout(Duration.ofSeconds(10))))
+                    .subscribe();
+            } catch (Exception ex) {
+                logger.error("Change feed processor did not start and stopped in the expected time", ex);
+                throw ex;
+            }
+
+            Thread.sleep(Duration.ofSeconds(20).toMillis());
+
+            ChangeFeedProcessor changeFeedProcessorSideCart = new ChangeFeedProcessorBuilder()
```

Comment:
We should add a comment here that the CFP lag checking can be performed on a separate application (like a health monitor) as long as the same input containers (feedContainer and leaseContainer) and the exact same lease prefix are used. The estimator code requires that the CFP had an opportunity to fully initialize the leaseContainer's documents were created first.